src/App/Header/UserMenu/UserMenu.tsx

```diff
@@ -0,0 +1,31 @@
+import { Authenticator, useAuthenticator } from "@aws-amplify/ui-react";
+import { Popover } from "@headlessui/react";
+import Icon from "components/Icon";
+import Menu from "./Menu";
+
+export default function UserMenu() {
+  const { user, signOut, route } = useAuthenticator((context) => [
+    context.route,
+  ]);
+
+  return (
+    <Popover className="relative">
+      {route === "authenticated" ? (
+        <Popover.Button className="cursor-pointer contents">
+          <Icon type="User" size={24} className="text-white" />
+        </Popover.Button>
+      ) : (
+        <Popover.Button className="btn-orange px-3 h-10 rounded-lg text-sm">
+          Sign in
+        </Popover.Button>
```

Comment:
When a new profile is created using the top right "Sign in" button, is it possible to add some kind of loading indicator after the email verification step? Not sure is amplify allows for such a thing, e.g. if it has an `isLoading` flag.

Repro video below. The loading indicator I'm asking about would be placed after the email verification step (at 0:14). Without this indicator it would seem to the user as if nothing is happening and then a couple of seconds later the "Sign in" button changes to the profile icon and they're suddenly signed in:
https://www.loom.com/share/620b74b2c42c4f05b160a27d7958d229?sid=b186d5a6-5637-4e23-a99d-64c77c7dbce4

Reply:
fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2427/commits/b84421be5352dbd1e00fd68794123013352930eb